docs/src/pages/jvm/index.mdx

```diff
@@ -6,64 +6,48 @@ export const showHero = true;
 
 <NavHeading text="literal" />
 
-Sure, you could write your assertions like this.
+This is a reasonable way to test.
 
 ```java
 @Test
-public void login() {
-  given().redirects().follow(false)
-    .post("/confirm/login/erjchFbCXxMlUfFXx3oYiO-Rj6zM7tRGdRV8ziqRn5Jh")
-    .then()
-    .cookie("login", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwiaXNzIjoiZGlmZnBsdWcuY29tIiwiYXVkIjoiZGlmZnBsdWcuY29tIiwiaWF0IjoxNTc3NjY0MDAsImVtYWlsIjoibHVrZS5za3l3YWxrZXJAZGlmZnBsdWcuY29tIiwic2FsZXMiOiJ0cnVlIn0.v7nes7_rTa0NiwTz6OLIAmRZJ-XzzGULRiAPWBiV5iI");
+public void primesBelow100() {
+  Assertions.assertThat(primesBelow(100)).startsWith(2, 3, 5, 7).endsWith(89, 97);
 }
 ```
 
-But isn't this easier to read? And also more complete?
+But oftentimes a more useful way to test is actually:
 
 ```java
 @Test
-public void login() {
-  expectSelfie(
-      given().redirects().follow(false).post("/confirm/login/erjchFbCXxMlUfFXx3oYiO-Rj6zM7tRGdRV8ziqRn5Jh").headers()
-    ).toBe("""
-      content-type=text/plain
-      set-cookie=flash=success=You+are+now+logged+in%21;Path=/;HttpOnly
-      set-cookie=loginui="{\"username\":\"testuser@diffplug.com\"}";Path=/;Max-Age=604800;Expires=Wed, 01-Jan-2000 00:00:00 GMT
-      set-cookie=login=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwiaXNzIjoiZGlmZnBsdWcuY29tIiwiYXVkIjoiZGlmZnBsdWcuY29tIiwiaWF0IjoxNTc3NjY0MDAsImVtYWlsIjoibHVrZS5za3l3YWxrZXJAZGlmZnBsdWcuY29tIiwic2FsZXMiOiJ0cnVlIn0.v7nes7_rTa0NiwTz6OLIAmRZJ-XzzGULRiAPWBiV5iI;Path=/;HttpOnly;Max-Age=604800;Expires=Wed, 01-Jan-2000 00:00:00 GMT
-      location=/settings
-      content-length=0
-      """);
+public void testMcTestFace() {
+  System.out.println(primesBelow(100));
 }
 ```
 
-The only reason we don't test that way already is that it's too tedious to write out that big string literal. Buf if you write this...
+With literal snapshots, you can `println` directly into your testcode, combining the speed and freedom of `println` with the repeatability and team friendliness of conventional assertions.
 
 ```java
 @Test
-public void login() {
-  expectSelfie(
-      given().redirects().follow(false).post("/confirm/login/erjchFbCXxMlUfFXx3oYiO-Rj6zM7tRGdRV8ziqRn5Jh").headers()
-    ).toBe_TODO()
+public void primesBelow100() {
+  expectSelfie(primesBelow(100).toString()).toBe_TODO();
 }
 ```
 
-... then selfie will rewrite the string literal for you when you run your tests.
-
-It's like a printf directly into your code. And it's not just for strings, it's for any literal.
+When you run the test, selfie will automatically rewrite `_TODO()` into whatever it turned out to be.
 
 ```java
 @Test
-public void preventCssBloat() {
-  int size = expectSelfie(get("/index.css").length).toBe(5_236);
-  if (size > 100_000) {
-    Assert.fail("CSS has gotten too big! " + size);
-  }
+public void primesBelow100() {
+  expectSelfie(primesBelow(100).toString())
+    .toBe("[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]");
```

Comment:
I think there's an opportunity for a good tie in to the next "like a filesystem" section.

I see a minor annoyance with an inline snapshot as big as this. It's going to make my test file gross and depending on my code formatter it could even make the test hard to read.

Cue next section... "don't want to pollute your test code? save it to disk!" I feel like it would be nice if the "like a filesystem" section started with this same example and then added on the binary example and finally the html example which itself further ties into the lensable section.

Now that's flow!

Reply:
good call https://github.com/diffplug/selfie/pull/95/commits/0a42cc0c9a69c49ce86e4976cc68f50b5133d059